flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/result/MaterializedCollectBatchResult.java

```diff
@@ -78,8 +82,9 @@ public boolean isMaterialized() {
 
 	@Override
 	public void startRetrieval(ProgramDeployer<C> deployer) {
-		this.deployer = deployer;
-		retrievalThread.start();
+		deployer.run()
```

Comment:
This changes the treading model of the class and all operations are executed by the same thread. Is this intentional?

Reply:
Yes, the change was on purpose. Before there was manual threading and waiting on a queue of results. Now this is replaced by futuristic code.